map/honor-mobile-map-package-expiration-date/src/main/java/com/esri/samples/honor_mobile_map_package_expiration_date/HonorMobileMapPackageExpirationDateSample.java

```diff
@@ -0,0 +1,153 @@
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // create an overlay to display the expiration information
+      VBox expirationMessageVbox = new VBox(6);
+      expirationMessageVbox.setBackground(new Background(new BackgroundFill(Paint.valueOf("rgba(0,0,0,0.3)"), CornerRadii.EMPTY,
+              Insets.EMPTY)));
+      expirationMessageVbox.setPadding(new Insets(10.0));
+      expirationMessageVbox.setMaxSize(800, 150);
+      expirationMessageVbox.setAlignment(Pos.CENTER);
+      expirationMessageVbox.getStyleClass().add("panel-region");
+
+      // create a label to display the expiration message and expiration date
+      Label expirationDetailsLabel = new Label();
+      expirationMessageVbox.getStyleClass().add("label");
+
+      // add the labels to the overlay
+      expirationMessageVbox.getChildren().add(expirationDetailsLabel);
+
+      // create a map view
+      mapView = new MapView();
+
+      // load the mobile map package
+      mobileMapPackage = new MobileMapPackage("./samples-data/mmpk/LothianRiversAnno.mmpk");
+      mobileMapPackage.loadAsync();
+      mobileMapPackage.addDoneLoadingListener(() -> {
+        // check if the map package has expiration information and if so, has it expired yet
+        if (mobileMapPackage.getExpiration() != null && mobileMapPackage.getExpiration().isExpired()) {
+
+          // get the expiration of the mobile map package
+          Expiration expiration = mobileMapPackage.getExpiration();
+
+          // get the expiration message
+          String expirationMessage = expiration.getMessage();
+
+          // get the expiration date
+          SimpleDateFormat daysHoursFormat = new SimpleDateFormat("EEE',' d MMM yyyy 'at' hh:mm:ss a", Locale.US);
+          String expirationDate = daysHoursFormat.format(expiration.getDateTime().getTimeInMillis());
+
+          // set the expiration message to the label
+          expirationDetailsLabel.setText(expirationMessage + "\n Mobile map package expired on: " + expirationDate + ".");
+
+          // load the expired map if it is still accessible after expiration
+          if (mobileMapPackage.getExpiration().getType() == ExpirationType.ALLOW_EXPIRED_ACCESS && mobileMapPackage.getLoadStatus() == LoadStatus.LOADED && !mobileMapPackage.getMaps().isEmpty()) {
+            // add the map from the mobile map package to the map view
+            mapView.setMap(mobileMapPackage.getMaps().get(0));
+
+            // show an alert if the mobile map package is not accessible after expiration
+          } else if (mobileMapPackage.getExpiration().getType() == ExpirationType.PREVENT_EXPIRED_ACCESS) {
+            new Alert(Alert.AlertType.ERROR, "The author of this mobile map package has disallowed access after the expiration date.").show();
+          }
+
+          // load the mobile map package if it has no expiration information
```

Comment:
you've already loaded it at this point. Maybe `//show the map if it is not expired`

Reply:
Well spotted, thanks!